src/services/chat/ws/hooks.ts

```diff
@@ -1,60 +1,52 @@
-// import {
-//   Actor,
-//   DatabaseTransactionHandler,
-//   ItemMembershipService,
-//   ItemService,
-//   TaskRunner,
-// } from '@graasp/sdk';
-// import { AccessDenied, NotFound, WebSocketService } from 'graasp-websockets';
+import { PermissionLevel } from '@graasp/sdk';
 
-// import { Chat } from '../interfaces/chat';
-// import { ChatMessage } from '../interfaces/chat-message';
-// import { ChatTaskManager } from '../interfaces/chat-task-manager';
-// import { ItemChatEvent, itemChatTopic } from './events';
+import { ItemNotFound } from '../../../utils/errors';
+import { Repositories } from '../../../utils/repositories';
+import ItemService from '../../item/service';
+import { WebsocketService } from '../../websockets/ws-service';
+import { ChatMessageService } from '../service';
+import { ItemChatEvent, itemChatTopic } from './events';
 
-// export function registerChatWsHooks(
-//   websockets: WebSocketService,
-//   runner: TaskRunner<Actor>,
-//   itemService: ItemService,
-//   itemMembershipService: ItemMembershipService,
-//   chatTaskManager: ChatTaskManager,
-//   validationDbHandler: DatabaseTransactionHandler,
-// ) {
-//   websockets.register(itemChatTopic, async (req) => {
-//     const { channel: itemId, member, reject } = req;
-//     // item must exist
-//     const item = await itemService.get(itemId, validationDbHandler);
-//     if (!item) {
-//       reject(NotFound());
-//     }
-//     // member must have at least read access to item
-//     const allowed = await itemMembershipService.canRead(member.id, item, validationDbHandler);
-//     if (!allowed) {
-//       reject(AccessDenied());
-//     }
-//   });
+export function registerChatWsHooks(
+  repositories: Repositories,
+  websockets: WebsocketService,
+  chatService: ChatMessageService,
+  itemService: ItemService,
+) {
+  websockets.register(itemChatTopic, async (req) => {
+    const { channel: itemId, member } = req;
+    // item must exist with read permission
+    const item = await itemService.get(member, repositories, itemId, PermissionLevel.Read);
+    if (!item) {
+      throw new ItemNotFound();
```

Comment:
Linked to my comment I think here the connection should simply be rejected with `reject(<the error>)` ? Maybe I am missing something, @dialexo can you help ?